central/sensorupgrade/service/service_impl_test.go

```diff
@@ -1,11 +1,175 @@
+			req: &v1.UpdateSensorUpgradeConfigRequest{
+				Config: &storage.SensorUpgradeConfig{EnableAutoUpgrade: true},
+			},
+			expectedErr:       errox.InvalidArgs,
+			upsertTimesCalled: 0,
+		},
+		"Success: can set toggle = false on managed centrals": {
+			managedCentral: true,
+			req: &v1.UpdateSensorUpgradeConfigRequest{
+				Config: &storage.SensorUpgradeConfig{EnableAutoUpgrade: false},
+			},
+			upsertTimesCalled: 1,
+		},
+		"Success: can set toggle = true on non-managed centrals": {
+			managedCentral: false,
+			req: &v1.UpdateSensorUpgradeConfigRequest{
+				Config: &storage.SensorUpgradeConfig{EnableAutoUpgrade: true},
+			},
+			upsertTimesCalled: 1,
+		},
+		"Success: can set toggle = false on non-managed centrals": {
+			managedCentral: false,
+			req: &v1.UpdateSensorUpgradeConfigRequest{
+				Config: &storage.SensorUpgradeConfig{EnableAutoUpgrade: false},
+			},
+			upsertTimesCalled: 1,
+		},
+	}
+
+	for caseName, testCase := range testCases {
+		s.Run(caseName, func() {
+			s.isolator.Setenv(env.ManagedCentral.EnvVar(), strconv.FormatBool(testCase.managedCentral))
+			s.dataStore.EXPECT().GetSensorUpgradeConfig(gomock.Any()).Times(1).Return(nil, nil)
+			s.dataStore.EXPECT().UpsertSensorUpgradeConfig(gomock.Any(), gomock.Any()).Times(1)
+			serviceInstance, err := New(s.dataStore, s.manager)
+			s.NoError(err)
+
+			s.dataStore.EXPECT().UpsertSensorUpgradeConfig(gomock.Any(), gomock.Eq(testCase.req.GetConfig())).
+				Times(testCase.upsertTimesCalled)
+			_, err = serviceInstance.UpdateSensorUpgradeConfig(context.Background(), testCase.req)
+			if testCase.expectedErr != nil {
+				s.ErrorIs(err, testCase.expectedErr)
+			} else {
+				s.NoError(err)
+			}
+
+			if testCase.upsertTimesCalled > 0 {
+				s.Require().Equal(serviceInstance.AutoUpgradeSetting().Get(),
+					testCase.req.GetConfig().GetEnableAutoUpgrade())
+			}
+		})
+	}
+}
+
+func (s *SensorUpgradeServiceTestSuite) Test_GetSensorUpgradeConfig_DefaultValues() {
+	testCases := map[string]struct {
+		expectedAutoUpdate     bool
+		expectedFeatureEnabled v1.GetSensorUpgradeConfigResponse_SensorAutoUpgradeFeatureStatus
+	}{
+		"true": {
+			expectedAutoUpdate:     false,
+			expectedFeatureEnabled: v1.GetSensorUpgradeConfigResponse_NOT_SUPPORTED,
+		},
+		"false": {
+			expectedAutoUpdate:     true,
+			expectedFeatureEnabled: v1.GetSensorUpgradeConfigResponse_SUPPORTED,
+		},
+	}
+
+	for envValue, expectations := range testCases {
+		s.Run(fmt.Sprintf("ROX_MANAGED_CENTRAL=%v", envValue), func() {
+			s.dataStore.EXPECT().GetSensorUpgradeConfig(gomock.Any()).Times(1).Return(nil, nil)
+			s.dataStore.EXPECT().UpsertSensorUpgradeConfig(gomock.Any(), &UpgradeConfigMatcher{expectations.expectedAutoUpdate})
+			s.isolator.Setenv(env.ManagedCentral.EnvVar(), envValue)
+
+			instance, err := New(s.dataStore, s.manager)
+			s.NoError(err)
+
+			s.dataStore.EXPECT().GetSensorUpgradeConfig(gomock.Any()).Times(1).Return(configWith(expectations.expectedAutoUpdate), nil)
```

Comment:
Maybe between these things validate that `instance.AutoUpgradeSetting().Get()` returns `expectations.expectedAutoUpdate` because that's why we need `initialize()`.

Reply:
Done